From 3cb4724e3e376a93f54e308bdf917932342680a2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 22 Nov 2017 19:50:41 +0000
Subject: [PATCH 3/4] Define two new helper macros

The PKG_ROOT_DIR will point to the root of the packages tree, which is
always one level HIGHER than the build directory.

The PKG_BUILD_DIR will point to the current build directory for the
build operation, i.e. `build`, `build-32`, which is always one level HIGHER
than the workdir.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/scripts.py     | 6 ++++++
 ypkg2/ypkgcontext.py | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/ypkg2/scripts.py b/ypkg2/scripts.py
index 1c7d378..27075d6 100644
--- a/ypkg2/scripts.py
+++ b/ypkg2/scripts.py
@@ -140,6 +140,9 @@ class ScriptGenerator:
         self.define_macro("version", self.context.spec.pkg_version)
         self.define_macro("sources", self.context.get_sources_directory())
 
+        self.define_macro("rootdir", self.context.get_package_root_dir())
+        self.define_macro("builddir", self.context.get_build_dir())
+
     def init_default_exports(self):
         """ Initialise our exports """
         self.exports = OrderedDict()
@@ -158,6 +161,9 @@ class ScriptGenerator:
         self.define_export("sources", "%sources%")
         self.define_export("pkgfiles", "%PKGFILES%")
         self.define_export("installdir", "%installroot%")
+        self.define_export("PKG_ROOT_DIR", "%rootdir%")
+        # Build dir, which is one level up from the source directory.
+        self.define_export("PKG_BUILD_DIR", "%builddir%")
         self.define_export("CC", self.context.build.cc)
         self.define_export("CXX", self.context.build.cxx)
         if self.context.build.ld_as_needed:
diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index aa98046..7835f39 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -246,6 +246,12 @@ class YpkgContext:
                                self.spec.pkg_name,
                                buildSuffix))
 
+    def get_package_root_dir(self):
+        """ Return the root directory for the package """
+        return os.path.abspath("{}/root/{}".format(
+                               self.get_build_prefix(),
+                               self.spec.pkg_name))
+
     def get_pgo_dir(self):
         """ Get the PGO data directory for the given package """
         pgoSuffix = "pgo"
-- 
2.15.0

